dev-setup.md

```diff
@@ -0,0 +1,186 @@
+  - Creates a guest of an appropriate size on the Host, and configures it appropriately
+  - Attempts to connect to the bootstrap node to ensure that it’s been built and started properly
+- 07\_deploy\_masters.sh
+  - Uses the OpenStack python-tripleoclient to create baremetal nodes to act as the masters
+  - A config drive is used by the nodes when they come up, which starts the process of installing the masters and ensuring that they can communicate with each other
+
+Two additional scripts are included as part of dev-scripts.  They will rarely need to be invoked directly, but are used by ‘make clean’ to clean up the host system:
+- ocp\_cleanup.sh
+  - Removes the bootstrap node along with its volumes, disk images, and associated ignition file
+  - Removes any residual OpenShift configuration data (found in the ocp/ directory)
+  - Removes the dnsmasq configuration created by 02\_configure\_host.sh
+
+- host-cleanup.sh
+  - Kills the Ironic podman containers running on the host
+  - Uses an ansible playbook to tear down all the masters and workers
+
+
+# Cleaning up your workspace with dev-scripts
+
+Cleanup is relatively simple by invoking the same ‘make’ process we used for setting up the environment in the first place.
+
+To clean up the entire environment, run:
+```
+$ make clean
+```
+
+# Developing against facet
+Up until this point, we’ve just set up the Host, the Masters, three Workers, and cloned the necessary repositories.  This has not started facet.
+
+Moving forward, the environment variable $GOPATH will be used to reference the Golang library path.  We can use the ‘go’ binary to create some environment variables in addition to $GOPATH:
+```
+$ eval `go env` 
+```
+
+The 03\_ocp\_repo\_sync.sh script we ran previously, automatically cloned the facet repo.  The repo now exists at $GOPATH/src/github.com/metalkube/facet.  This is necessary for the default facet production build, but needs additional work for active development.
+
+To run the production build of facet, use the following command:
+```
+$ go run $GOPATH/src/github.com/metalkube/facet/main.go server
+```
+
+Upon execution, facet can be accessed by the URL printed to the output.
+
+From here, facet development can be expanded by using the documentation, and standard development practices apply moving forward.
+
+
+# Q/A
+
+**Q**:  How do I make facet listen on an IP / host other than ‘localhost’?  
+**A**:  It might be desirable to change the listening address of the Go server.  To do so, edit pkg/server/server.go around like 53, replacing the default ‘localhost’ with a different IP address, or 0.0.0.0 for all IPs.  
+
+**Q**:  How do I connect to the masters?  
+**A**:  Connecting to the masters shouldn’t be necessary, but in a pinch, you can access them by ssh core@master-<index>.api.ostest.test.metalkube.org  
```

Comment:
hmm, something is wrong here, when I tested it all of these names resolved to the same IP (the VIP), I was messing around with my env, going to retry in new env.

Reply:
I don't have a working environment at the moment to test this for myself.  I'll follow up with this later today when I do.  This might have been a change as a side effect of the mDNS and dnsmasq change we made week before last.

Reply:
I have a fix waiting to go in that makes dns via libvirtd's dnsmasq work properly.  I think at the time of this writing, I was also working on this.  I'll have the PR for this fix in shortly but it will work.  The mDNS work doesn't look like it has an effect on this, either, since the usability this note relies on operates off of the host dnsmasq instance of libvirt.